### PR TITLE
SOLR-16005 added new feature in to schema screen: add new field type

### PR DESCRIPTION
**Pull request by ackepenek**
https://issues.apache.org/jira/browse/SOLR-16005

# Description

Added "add field type" feature in to collection's schema screen.

# Solution

The button and functionality is added like add field. 

# Tests

field type addition is tested manually.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

screenshot about improvement:

image

### CODE REVIEW

**Comment by ackepenek** [comment]
> Thanks @ackepenek, left a review. How would you like to be credited in the changelog???

Thanks for review. "Ahmet Can Kepenek" is good to be in changelog.

**Comment by epugh** [comment]
@ackepenek so I tried out the feature locally, and I really like it, but did have a thought....   Instead of this being a hard coded to the `add-field-type` command, what if instead it was an interface for you to paste your JSON commands?   The use case that I was thinking about is that as I iterate on my relevance algorithem, I'll often modify the same field a couple of times as I figure out what I need, which means I either have a script that does a `delete-field-type` and then `add-field-type` or I used the `replace-field-type`, though honestly I mostly just Delete and Add!

What if the button said "Manipulate Field Type" and then it took any json, and ran it, so it assumes you pass in `add-field-type`, `replace-field-type` or `delete-field-type` as appropriate?    That might make this feature a lot more robust all the way around, and avoid you having to add a "Delete" button?

Most people using this are probably crafting the JSON outside anyway. 

Also, if this button works well here, I wonder if something like it should be added to the Schema Designer?

THoughts?

**Comment by ackepenek** [comment]
I improved the PR by comments. 

latest screenshot:

image

**Comment by epugh** [comment]
Thank you for looking @madrob and @janhoy...    So the first version that @ackepenek had was basically the "Add field type Template".   You just put in the JSON, without the `add-field-template` command.   My gut feeling is that this is a tool that people coming from using Postman or other JSON syntax native tools, who don't want to figure out how to craft the perfect URL, would benefit from.   Those people would already have the command in it...

This does make me wonder, is the real utility of this screen that you can take your JSON commands and just paste them in and run them wihtout crafting the URL, or figuring out any security etc because you are in the admin tool?   

Maybe what we really want is to support what https://solr.apache.org/guide/8_11/schema-api.html#multiple-commands-in-a-single-post suggests.   What if this was the "Scripted Schema Modifications"?   We reference the schema api docs, and you can mix up your various commands, `add-field-type`, `add-field`, `delete-field` etc?  They all go through the same `Schema.post` command anyway, right?  

This would let me take my `setup-experiment-schema.json` file that i already might have, and not have to worry about postman etc, just cut n paste right into solr.    It also would be more clearly not a "Here is the simple accessible UI"...

**Comment by ackepenek** [comment]
@madrob @janhoy @epugh 
I am agree with @epugh. We should get better all admin panel to understand what we can do. We should put info text to panel where we need. I think it is makes more sense to do analyze to improve admin panel usage.
We just give a capability to manipulate the field type. It's better than zero capability. I think this development is simple and useful.

 do we have any concerns still? or any requests? I am open any ideas.

**Comment by epugh** [comment]
I like your suggestion @janhoy and that would get this to done.   @ackepenek what do you think?

**Comment by ackepenek** [comment]
> I like your suggestion @janhoy and that would get this to done. @ackepenek what do you think?

It is just a decision. We can think to make better panel considering all features later💫

For now I will improve the feature like Jan said. Let me now the change the PR 👍

**Comment by ackepenek** [comment]
@epugh I fixed the PR as you mentioned. You can see new screenshots below:


Screenshot from 2022-07-06 04-16-48
Screenshot from 2022-07-06 04-17-10

File: solr/webapp/web/partials/schema.html [hunk 1]
```diff
@@ -218,10 +218,10 @@
...
                     </p>
                     <p>
-                        <a target="_blank" href="https://solr.apache.org/guide/solr/latest/indexing-guide/schema-api.html#add-a-new-field-type">Delete a Field Type</a>: The <b>delete-field-type</b> command removes a field type from your schema.
+                        <a target="_blank" href="https://solr.apache.org/guide/solr/latest/indexing-guide/schema-api.html#delete-a-new-field-type">Delete a Field Type</a>: The <b>delete-field-type</b> command removes a field type from your schema.
                     </p>
                     <p>
-                        <a target="_blank" href="https://solr.apache.org/guide/solr/latest/indexing-guide/schema-api.html#add-a-new-field-type">Replace a Field Type</a>: The <b>replace-field-type</b> command replaces a field type in your schema.
+                        <a target="_blank" href="https://solr.apache.org/guide/solr/latest/indexing-guide/schema-api.html#replace-a-new-field-type">Replace a Field Type</a>: The <b>replace-field-type</b> command replaces a field type in your schema.
```
Comment (ackepenek) [inline solr/webapp/web/partials/schema.html:224]: ohh my fault 😞 thanks :pray:

**Comment by epugh** [comment]
down tjhe road, we can get rid of the JSON and make it all pure admin gui.

**Comment by janhoy** [comment]
Just as I hoped, LGTM. Cannot test (holiday).

**Pull request merged**